src/bootstrap/src/core/builder.rs

```diff
@@ -72,36 +72,40 @@ impl<'a> Deref for Builder<'a> {
 }
 
 pub trait Step: 'static + Clone + Debug + PartialEq + Eq + Hash {
-    /// `PathBuf` when directories are created or to return a `Compiler` once
-    /// it's been assembled.
+    /// Result type of `Step::run`.
     type Output: Clone;
 
     /// Whether this step is run by default as part of its respective phase.
-    /// `true` here can still be overwritten by `should_run` calling `default_condition`.
+    ///
+    /// Note: Even if set to `true`, it can still be overridden (with `default_condition`) by `should_run`.
     const DEFAULT: bool = false;
```

Comment:
Question: as someone who sometimes work with `Step`s but am not familiar with all the boostrap internals, I don't understand:

- What "respective phase" means here. Is it like the `Build::get_step_descriptions` -> `kind::Foo` -> `describe!( tool::Foo )`?
- What `default_condition` is supposed to mean or indicate here?

Could you kindly elaborate? Or link to other docs that explain what these is. I don't want to accidentally add subtle bugs when implementing `Step`s lol.

Reply:
> What "respective phase" means here. Is it like the Build::get_step_descriptions -> kind::Foo -> describe!( tool::Foo )?

Exactly.

> What default_condition is supposed to mean or indicate here?

It's `ShouldRun::default_condition`, you can see how it's utilized in other steps. [Updated](https://github.com/rust-lang/rust/pull/130965/commits/33614fdabcd2d4bab19ea328d44cf6f24a4565e8) the doc again for some more details for making it more clear.

Reply:
Thanks for the clarification!